sympy/functions/elementary/complexes.py

```diff
@@ -442,7 +453,7 @@ def eval(cls, arg):
             known = []
             unk = []
             for t in arg.args:
-                tnew = cls(t)
+                tnew = refine(cls(t))
```

Comment:
I'd say you should, although you may want to rewrite the whole function using `Mul.make_args` so that it does work when `a` is a number.